lib/test_helpers/pdf_comparison_test_factory.rb

```diff
@@ -0,0 +1,146 @@
+    create_profile(partner: partner, program_address1: "Example Program Address 1", program_address2: "", program_city: "Example Program City", program_state: "Example Program State", program_zip: 54321)
+  end
+
+  def create_profile_with_incomplete_address(partner)
+    create_profile(partner: partner, program_address1: "Example Program Address 1", program_address2: "", program_city: "", program_state: "", program_zip: "")
+  end
+
+  def create_profile_no_contact_with_program_address(partner)
+    create_profile(partner: partner, program_address1: "Example Program Address 1", program_address2: "", program_city: "Example Program City", program_state: "Example Program State", program_zip: 54321, primary_contact_name: "", primary_contact_email: "")
+  end
+
+  def create_line_items_request(distribution, partner, storage_creation)
+    LineItem.create!(itemizable: distribution, item: storage_creation.items[0], quantity: 50)
+    LineItem.create!(itemizable: distribution, item: storage_creation.items[1], quantity: 100)
+    storage_creation.organization.request_units.find_or_create_by!(name: "pack")
+    ItemUnit.find_or_create_by!(item: storage_creation.items[3], name: "pack")
+    req1 = Partners::ItemRequest.new(item: storage_creation.items[1], quantity: 30, name: storage_creation.items[1].name, partner_key: storage_creation.items[1].partner_key)
+    req2 = Partners::ItemRequest.new(item: storage_creation.items[2], quantity: 50, name: storage_creation.items[2].name, partner_key: storage_creation.items[2].partner_key)
+    req3 = Partners::ItemRequest.new(item: storage_creation.items[3], quantity: 120, name: storage_creation.items[3].name, partner_key: storage_creation.items[3].partner_key, request_unit: "pack")
+    Request.create!(
+      organization: storage_creation.organization,
+      partner: partner,
+      distribution: distribution,
+      request_items: [
+        {"item_id" => storage_creation.items[1].id, "quantity" => 30},
+        {"item_id" => storage_creation.items[2].id, "quantity" => 50},
+        {"item_id" => storage_creation.items[3].id, "quantity" => 120, "request_unit" => "pack"}
+      ],
+      item_requests: [req1, req2, req3]
+    )
+  end
+
+  def create_dist(partner, storage_creation, delivery_method)
+    Time.zone = "America/Los_Angeles"
+    dist = Distribution.create!(partner: partner, delivery_method: delivery_method, issued_at: DateTime.new(2024, 7, 4, 0, 0, 0, "-07:00"), organization: storage_creation.organization, storage_location: storage_creation.storage_location)
+    create_line_items_request(dist, partner, storage_creation)
+    dist
+  end
+
+  private def create_comparison_pdf(storage_creation, profile_create_method, expected_file_path, delivery_method)
+    partner = create_partner(storage_creation.organization)
+    profile = PDFComparisonTestFactory.instance_method(profile_create_method).bind_call(Class.new.extend(PDFComparisonTestFactory), partner)
```

Comment:
This is a bit over-the-top... can we not just move the possible inputs to this to class methods? Then you can just call `PDFComparisonTestFactory.send(profile_create_method)`.